samples/c/get_setup_mode.c

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: LGPL-2.1-only
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates
+ *
+ * Author: Kamalesh Babulal <kamalesh.babulal@oracle.com>
+ *
+ * Description: This file contains the sample code to demonstrate usage of
+ * 		cgroup_setup_mode() API.
+ */
+
+#include "../src/libcgroup-internal.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+int main(void)
+{
+	enum cg_setup_mode_t setup_mode;
+	int ret;
+
+	ret = cgroup_init();
+	if (ret) {
+		printf("cgroup_init failed with %s\n", cgroup_strerror(ret));
+		exit(1);
+	}
+
+	setup_mode = cgroup_setup_mode();
+	switch(setup_mode) {
+	case 1:
```

Comment:
`CGROUP_MODE_LEGACY` is available via the public header `libcgroup.h`.  Shouldn't a user use it rather than the magic number `1`?